tests/test_solver.py

```diff
@@ -31,17 +30,19 @@ def heat_balance(conductor_temperature, current):
     assert conductor_temperature == pytest.approx(9000, rel=1e-7)
 
 
-def test_compute_conductor_temperature_caps_ampacity_at_zero():
+def test_compute_conductor_temperature_raises_value_error():
```

Comment:
It would be nice to have a test showing what happens when we have numpy array output from the heat_balance function, I couldn't see other tests that covers the array case. For example it would be nice to have a test documenting what happens when one element in an array has the same sign for the callable, i.e. for an array `np.array([x_0, x_1])` where `sign(f(xmax_0)) == sign(f(xmin_0))` but `sign(f(xmax_1)) != sign(f(xmin_1))`

Reply:
Added three tests for this:
test_bisect_handles_function_returning_array_happy_path
test_bisect_raises_valueerror_when_same_sign_for_array_input
test_bisect_raises_valueerror_when_infinite_in_array_input